src/main/controller/PromotionController.ts

```diff
@@ -204,4 +196,70 @@ export class PromotionController {
       return next(e);
     }
   };
+
+  /**
+   * Get the restaurant details for a promotion
+   */
+  getRestaurantDetails = async (
+    request: Request,
+    response: Response,
+    next: NextFunction
+  ): Promise<any> => {
+    try {
+      await getManager().transaction(async (transactionalEntityManager) => {
+        let result: Place = {};
+        const id = await IdValidation.schema.validateAsync(request.params.id, {
+          abortEarly: false,
+        });
+
+        const promotion = await transactionalEntityManager
+          .getCustomRepository(PromotionRepository)
+          .findOneOrFail(id);
+
+        // placeId may be empty string if this promotion has previously resulted in a NOT_FOUND even after refresh request
+        if (promotion.placeId) {
+          const placeDetailsResponseData = await this.googlePlaceService.getRestaurantDetails(
+            promotion.placeId
+          );
+          result = placeDetailsResponseData.result ?? {};
+
+          if (placeDetailsResponseData.status === Status.NOT_FOUND) {
+            result = await this.handlePlaceIdNotFound(
+              promotion.placeId,
+              id,
+              transactionalEntityManager
+            );
+          }
+        }
+
+        return response.status(200).send(result);
```

Comment:
If the `promotion.placeId` is an empty string, we return an empty result?

Reply:
Yup correct. We can spend a larger effort in future to account for placeId being empty string, which only happens if refresh request fails